guide_drafts/composing-diesel-applications.md

```diff
@@ -0,0 +1,246 @@
+Composing Applications with Diesel
+----------------------------------
+
+One of the main benefits of using a query builder over raw SQL
+is that you can pull bits of your query out into functions and reuse them.
+In this guide,
+we'll look at common patterns for extracting your code into re-usable pieces.
+We'll also look at best practices for how to structure your code.
+
+All of our code examples are based on code from crates.io,
+a real world application which uses Diesel extensively.
+All of our examples will be focused on functions which *return*
+queries or pieces of queries.
+None of these examples will include a function which takes a database
+connection.
+We will go into the benefits of this structure at the end of the guide.
+
+crates.io has a `canon_crate_name` SQL function
+which is always used when comparing crate names.
+Rather than continuously writing
+`canon_crate_name(crates::name).eq("some name")`,
+we can instead pull this out into a function.
+
+```
```

Comment:
As noted on gitter: This should have a language hint.

Reply:
I dont bother adding them for drafts. I have to do a ton of work on code blocks when I publish